### DISCUSSION TITLE
Can I obtain RID(residual interstory disp) in R2D?

### DISCUSSION THREAD

**Discussion by pby22555** [NHERI-SimCenter/SimCenterCommon#215] [R2D]
Dear developer, can I obtain value of residual_disp in R2D?
![image](https://github.com/user-attachments/assets/77330c2f-71c5-479c-bb41-a2f47e616d42)

**Comment by pby22555** [comment]
Dear friend, how can I obtain RID(residual interstory disp) in R2D?

**Comment by jinyan1214** [comment]
R2D does not output RID. You'll need to modify the backend code to get it. If you can elaborate the modeling and analysis procedure you plan to perform on R2D, we can provide more guidance.

**Reply by pby22555** [reply]
Dear Jinyan,

I hope this email finds you well. I am currently working on replacing traditional buildings with emerging structures, such as self-centering buildings, to evaluate their seismic performance. Residual interstory displacement (RID) is a critical parameter for self-centering buildings.

I noticed that there is a "userEDP" application located in the "R2D_Windows_Download\applications\createEDP" folder, and I also found references to "RID" in the "SurrogateSimulation.py" file. Based on this, I attempted to modify the "EDP": {"Buildings": {"Application": "userEDP", ...}} section in the "input.json" file, hoping to extract the RID. Unfortunately, I have not been successful so far.

Could you please guide me on how to obtain the RID? If possible, could you demonstrate using the "E3PhysicsBasedGroundMotions" example or another example from the manual? This would greatly aid my research.

Thank you very much for your assistance！！

Sincerely,
Boyan

**Reply by jinyan1214** [reply]
Hi Boyan,

This is a very interesting application of R2D. I am very happy to support you on this. Can you send me an email at jinyan_zhao@berkeley.edu, so that I can ask for more details and perhaps arrange a zoom meeting to talk about it?

Thanks and regards.

**Reply by pby22555** [reply]
Dear Jinyan,

I have already sent an email, and my email address is 1025902088@qq.com.

Sincerely,
Boyan
